### PR TITLE
PIM-6258: fix permission issue

### PR DESCRIPTION
**Pull request by solivier**
While accessing the PEF, the UI calls rest routes for listing locales, associations, families and more. If there are no permission on this groups it returns a 403.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | No
| Added Behats                      | Yes
| Added integration tests           | No
| Changelog updated                 | Yes
| Review and 2 GTM                  | Todo

### CODE REVIEW

**Comment by skeleton** [comment]
it means everybody can access to family, channel, etc whatever ACLs ?

File: CHANGELOG-1.7.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 - PIM-6239: Translate scope with catalog locale
 - GITHUB-3435: Sort order products datagrid `Manage filter` options
 - GITHUB-5538: User without permissions access to import/export jobs through `Process tracker`
+- PIM-6258: Fix permission issue
```
Comment (anaelChardan) [inline CHANGELOG-1.7.md]: Permission about what ?

File: features/product/edit_product.feature [hunk 2]
```diff
@@ -100,3 +100,15 @@ Feature: Edit a product
...
+  @jira https://akeneo.atlassian.net/browse/PIM-6258
+  Scenario: Successfully view a product even if permissions on locales channels and families are revoked
+    Given I am on the "Catalog manager" role page
+    When I visit the "Permissions" tab
+    And I revoke rights to group Association types
+    And I revoke rights to group Channels
+    And I revoke rights to group Families
+    And I revoke rights to group Locales
+    And I save the role
+    Then I edit the "rangers" product
```
Comment (anaelChardan) [inline features/product/edit_product.feature]: And :-)

File: features/product/edit_product.feature [hunk 3]
```diff
@@ -100,3 +100,15 @@ Feature: Edit a product
...
+    And I should see the text "Rangers"
```
Comment (anaelChardan) [inline features/product/edit_product.feature]: Then :)

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/AssociationTypeController.php [hunk 4]
```diff
@@ -50,6 +50,11 @@ class AssociationTypeController
     /**
      * @param AssociationTypeRepositoryInterface $associationTypeRepo
      * @param NormalizerInterface                $normalizer
+     * @param RemoverInterface                   $remover
+     * @param ObjectUpdaterInterface             $updater
+     * @param SaverInterface                     $saver
+     * @param ValidatorInterface                 $validator
+     * @param UserContext                        $userContext
```
Comment (anaelChardan) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/AssociationTypeController.php]: +1 :)

**Comment by solivier** [comment]
@skeleton no it's only to list locales, associations, families and more. It's called on the rest controller/

File: CHANGELOG-1.7.md [hunk 5]
```diff
@@ -10,6 +10,7 @@
 - GITHUB-3435: Sort order products datagrid `Manage filter` options
 - PIM-6250: Fix attribute export.
 - GITHUB-5538: User without permissions access to import/export jobs through `Process tracker`
+- PIM-6258: Fix permissions issue for listing locales, associations, families to display correctly the pef
```
Comment (phaseinducer) [inline CHANGELOG-1.7.md]: product edit form

**Comment by a2xchip** [comment]
Ok! We have to remove ancestors annotations like `@AclAncestor("pim_enrich_associationtype_index")` but we can check perms inside like ```if (!$security->isGranted('pim_enrich_associationtype_index') && !$security->isGranted('pim_product_show')) trhow bla bla`

**Comment by a2xchip** [comment]
:disappointed_relieved: :cry:

**Pull request merged**